### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by rajachan**
See individual commit messages for details.

Some more updates are needed to the README.md, which will come in a following PR.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bwbarrett** [comment]
Sorry, I didn't see this before I did some long standing doc update work.  We shouldn't treat master docs like release docs, because it confuses our customers and leaders.  But let's target this patch (minus the code owners, which is better than mine and I'll drop mine) to the v1.8.x branch you're about to create.

File: INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Building the AWS OFI NCCL Plugin
+
+We strongly recommend starting with a release tarball available on the
+[GitHub Release Page](https://github.com/aws/aws-ofi-nccl/releases) when
+building from source for production uses..
```
Comment (bwbarrett) [inline INSTALL.md:5]: double period.

File: INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+`aws-ofi-nccl` requires a working installation of Libfabric. You can
```
Comment (bwbarrett) [inline INSTALL.md:7]: we should specify a version here.  And note that the version required is different on EFA than other providers.

**Review by bwbarrett — CHANGES_REQUESTED** [review]
Looks good, other than retarget to the 1.8.x branch.

File: INSTALL.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+$ sudo make install
+```
+
+If you want to install the plugin in a custom path, use the `--prefix`
+configure flag to provide the path. You can also point the build to custom
+dependencies with the following flags:
+
+```
+  --with-libfabric=PATH   Path to non-standard libfabric installation
+  --with-cuda=PATH        Path to non-standard CUDA installation
```
Comment (rauteric) [inline INSTALL.md:26]: Shouldn't we update this, as the path is ignored now?

**Comment by rajachan** [comment]
Closing to rework on top of #335

**Pull request closed without merging**